### PR TITLE
Import testify/mock replaced with gomock

### LINKED ISSUE #26: Use Go Mock
## Problem
Mocks used for testing are currently written manually using [testify/mock](github.com/stretchr/testify/mock). While this provides necessary functionality for testing, it doesn't scale well for the addition of new testing mocks and regeneration of existing ones.

## Problem Solution
[GoMock](https://github.com/golang/mock) provides mock autogeneration capabilities using embed instructions within Go files. 

### Acceptance Criteria
I. All current mocks (i.e, `EthClientInterface`) are regenerated using `mockgen` and stored as independent files in an `internal/mocks` package

II. All current mocks have embedded `gomock` instructions at the top of the code files which declare the interface that we'd like to mock. For example, to mock `EthClientInterface`, we add the following line to the top of `internal/client/eth_client.go`:
```
//go:generate mockgen -package=mocks -destination=../mocks/eth_client.go github.com/base-org/pessimism/internal/client EthClientInterface

```

III. Mock generation should be possible using a make command. _(e.g, `make mock-gen`)_

### PR DESCRIPTION
**Pull request by vavrajosef**
Closes #26

Hi, I have replaced testify/mock with gomock. EthClient is autogenerated with mockgen.
Please let me know if the tests are rewritten correctly.

### CODE REVIEW

**Comment by epociask** [comment]
Hi @vavrajosef, just wanna say thank you greatly for taking the time to open this PR. @anupsv-cb will validate the test implementations, in the meantime I've dropped some feedback and have formally assigned the issue to you :)

File: internal/mock_client/eth_client.go
Comment (epociask) [inline internal/mock_client/eth_client.go:4]: Any chance we could instead store this mock file in an `internal/mocks` package that can be used to durably support additional generated mocks without convoluting the internal file structure?

File: internal/conduit/registry/geth_block_test.go
Comment (epociask) [inline internal/conduit/registry/geth_block_test.go:11]: @anupsv-cb Any chance you could look over this file?

**Review by epociask — COMMENTED** [review]
Dropped some feedback, thank you again for taking this on. In the meantime, I've also updated the issue (#26) with a proper acceptance criteria that can hopefully provide better context as well. Can you please comment on the issue so that I can properly assign it to you?

File: internal/conduit/registry/geth_block_test.go
Comment (anupsv-cb) [reply internal/conduit/registry/geth_block_test.go:11]: Definetely.

File: internal/mock_client/eth_client.go
Comment (anupsv-cb) [reply internal/mock_client/eth_client.go:4]: +1

Comment (vavrajosef) [reply internal/mock_client/eth_client.go]: I moved the file to mocks and updated Makefile, so anytime make test is run, the mock is generated.

File: Makefile
Comment (epociask) [inline Makefile:26]:
```suggestion
.PHONY: go-gen-mocks
go-gen-mocks:
	@echo "generating go mocks..."
	@GO111MODULE=on go generate --run "mockgen*" ./...
	
.PHONY: test
test:
	@ go test ./... -v -timeout $(TEST_LIMIT)
```

**Review by epociask — COMMENTED** [review]
Just one comment, otherwise LGTM. Can't approve until you fix merge conflicts :)

**Comment by vavrajosef** [comment]
@epociask Merge conflicts should be resolved.

**Comment by epociask** [comment]
Looks like it's failing lint: https://github.com/base-org/pessimism/actions/runs/4751519710/jobs/8453021423?pr=27 

You can verify lint compliance locally by running:
```
make lint
```

Apologies for not having instructions for this yet, we've only been rolling for a few weeks now!

**Comment by anupsv-cb** [comment]
> Looks like it's failing lint: https://github.com/base-org/pessimism/actions/runs/4751519710/jobs/8453021423?pr=27
> 
> You can verify lint compliance locally by running:
> 
> ```
> make lint
> ```
> 
> Apologies for not having instructions for this yet, we've only been rolling for a few weeks now!

+1

Rest looks good

**Comment by vavrajosef** [comment]
I am using windows, so I had some issues with golint. I think I fixed everything that was reported, but when someone runs it here, we will know for sure.

**Review by anupsv-cb — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for anupsv-cb @ 2023-04-21 21:16:29 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by anupsv-cb — APPROVED** [review]

**Pull request merged**
